apps/emqx/src/emqx_connection.erl

```diff
@@ -705,6 +716,17 @@ serialize_and_inc_stats_fun(#state{serialize = Serialize}) ->
             Data -> ?LOG(debug, "SEND ~s", [emqx_packet:format(Packet)]),
                     ok = inc_outgoing_stats(Packet),
                     Data
+        catch
+            %% Maybe Never happen.
+            throw : ?FRAME_SERIALIZE_ERROR(Reason) ->
+                ?SLOG(info, #{ reason => Reason
+                             , input_packet => Packet}),
+                erlang:error(?FRAME_SERIALIZE_ERROR(Reason));
+            error : Reason : Stacktrace ->
+                ?SLOG(error, #{ input_packet => Packet
+                              , exception => Reason
+                              , stacktrace => Stacktrace}),
+                erlang:raise(error, Reason, Stacktrace)
```

Comment:
```suggestion
                erlang:error(frame_serialization_error)
```
the error reason and stacktrace are already logged, so we can simplify the error context for upper level.
raising with the exact same error context will most likely cause duplicated logging.